### PR TITLE
add code editor

### PR DESCRIPTION
**Pull request by hongl-1**
### What is the purpose of this pull request? 

- [ ] Bug fix
- [x] New Feature
- [ ] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [ ] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -2,10 +2,18 @@ lockfileVersion: 5.3
 
 specifiers:
   '@ant-design/icons-vue': ^6.1.0
+  '@codemirror/basic-setup': ^0.20.0
+  '@codemirror/commands': ^0.20.0
+  '@codemirror/gutter': ^0.19.9
+  '@codemirror/highlight': ^0.19.8
+  '@codemirror/lang-css': ^0.20.0
+  '@codemirror/lang-javascript': ^0.20.0
+  '@codemirror/view': ^0.20.3
```
Comment (Hongbusi) [inline pnpm-lock.yaml:11]: 猜测可能是这里导致的报错问题。
我在本地执行 install 之后这里是被移除了的。

File: src/views/demo/code-editor/index.vue [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+]
+
+function onChangeTheme(theme: string) {
+  message.success(`操作成功!当前主题为:${theme}`)
+}
+function onChangeLang(lang: string) {
+  message.success(`操作成功!当前主题为:${lang}`)
+}
+function onCodeChange(code: string) {}
+</script>
```
Comment (Hongbusi) [inline src/views/demo/code-editor/index.vue:75]: - 可以用一下 page-wrapper 组件；
- 是不是可以多添加几种语言支持？
- 默认宽度占容器的百分百会不会好看一些？

File: src/components/CodeEditor/src/CodeEditor.vue [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+<script setup lang="ts">
+import 'codemirror/theme/ambiance.css'
+import 'codemirror/lib/codemirror.css'
+import 'codemirror/addon/hint/show-hint.css'
+import 'codemirror/addon/edit/matchbrackets'
+import 'codemirror/addon/selection/active-line'
+import 'codemirror/addon/hint/show-hint'
+import CodeMirror from 'codemirror/lib/codemirror'
+import { onMounted, ref } from 'vue'
```
Comment (Hongbusi) [inline src/components/CodeEditor/src/CodeEditor.vue:18]: 已经使用 unplugin-auto-imports 全局导入了

File: pnpm-lock.yaml (hunk 1 above)
Comment (hongl-1) [reply pnpm-lock.yaml:11]: 这个我处理一下 我之前在package中手动删除的

File: src/components/CodeEditor/src/CodeEditor.vue (hunk 3 above)
Comment (hongl-1) [reply src/components/CodeEditor/src/CodeEditor.vue:18]: ok

File: src/views/demo/code-editor/index.vue (hunk 2 above)
Comment (hongl-1) [reply src/views/demo/code-editor/index.vue:75]: page-warpper 我加一下
语言是都可以支持的 只是我现在demo里面只展示了2种 我准备后续完善在文档说明中

**Comment by hongl-1** [comment]
重新提交了一个pr #79

**Pull request closed without merging**